exonum/src/runtime/mod.rs

```diff
@@ -120,53 +122,118 @@ impl From<RuntimeIdentifier> for u32 {
     }
 }
 
-/// Runtime environment for services.
+/// This trait describes runtime environment for the Exonum services.
 ///
-/// It does not assign id to services/interfaces, ids are given to runtime from outside.
+/// You can read more about the life cycle of services and transactions
+/// [above](index.html#service-life-cycle).
+///
+/// Using this trait, you can extend the Exonum blockchain with the services written in
+/// different languages. It assumes that the deployment procedure of a new service may be
+/// complex and long and even may fail;
+/// therefore, it was introduced an additional entity - artifacts.
+/// Each artifact has a unique identifier and, depending on the runtime, may have an additional
+/// specification which needs for its deployment. For example, the file to be compiled.
+/// Artifact creates corresponding services instances, the same way as classes in object
+/// oriented programming.
+///
+/// Please pay attention to the panic handling policy during the implementation of methods.
+/// If no policy is specified, then the method should not panic and each panic will abort node.
+///
+/// Keep in mind that runtime methods can be executed in two ways: during the blocks execution
+/// and during the node restart, thus be careful not to do unnecessary actions in the runtime
+/// methods.
+///
+/// # Hints
+///
+/// * You may use [`catch_panic`](error/fn.catch_panic.html) method to catch panics in order of panic policy.
 pub trait Runtime: Send + Debug + 'static {
     /// Request to deploy artifact with the given identifier and additional specification.
-    /// It immediately returns true if artifact have already deployed.
+    ///
+    /// # Policy on panics
+    ///
+    /// * This method should catch each kind of panics except of `FatalError` and converts
+    /// them into `ExecutionError`.
     fn deploy_artifact(
         &mut self,
         artifact: ArtifactId,
         spec: Any,
     ) -> Box<dyn Future<Item = (), Error = ExecutionError>>;
 
-    /// Returns additional information about artifact with the specified id if it is deployed.
+    /// Returns protobuf description of deployed artifact with the specified identifier.
```

Comment:
I'd add what Option#empty means — no artifact found by the given id; or that artifacts might not have ArtifactInfo.

Reply:
Each deployed artifact should has artifact information.